lib/feature_flagger/notifier.rb

```diff
@@ -0,0 +1,47 @@
+module FeatureFlagger
+  class Notifier
+    attr_reader :notify
+
+    RELEASE = 'release'.freeze
+    UNRELEASE = 'unrelease'.freeze
+    RELEASE_TO_ALL = 'release_to_all'.freeze
+    UNRELEASE_TO_ALL = 'unrelease_to_all'.freeze
+
+    def initialize(notify)
+      @notify = check_notify(notify)
+    end
+
+    def send(operation, feature_key, resource_id = nil)
+      @notify.call(build_event(operation, extract_resource_from_key(feature_key), feature_key, resource_id)) if notify?
+    end
+
+    private
+
+    def check_notify(notify)
+      return nil if notify.nil?
+    raise ArgumentError, "Notifier callback should be a lambda" unless notify.is_a?(Proc)
+      notify
+    end
```

Comment:
Doing this way the user always needs to define the notifier callback. What do you think about to make the callback optional?